pkg/cmd/experimental/buildchain/buildchain.go

```diff
@@ -94,12 +94,24 @@ func (o *BuildChainOptions) Complete(f *clientcmd.Factory, cmd *cobra.Command, a
 	}
 	o.c, o.t = oc, oc
 
-	// Parse user input
-	o.name, o.tag, err = buildChainInput(args[0])
+	resource := ""
+	mapper, _ := f.Object()
+	resource, o.name, err = osutil.ResolveResource("imagestreamtags", args[0], mapper)
 	if err != nil {
-		return cmdutil.UsageError(cmd, err.Error())
+		return err
+	}
+
+	switch resource {
+	case "imagestreamtags", "imagestreamtag":
+	default:
+		return fmt.Errorf("invalid resource provided: %s", resource)
 	}
-	glog.V(4).Infof("Using '%s:%s' as the image stream tag to look dependencies for", o.name, o.tag)
+
+	if !strings.Contains(o.name, ":") {
```

Comment:
Move this into the `case "imagestreamtags":` above, since this only makes sense for imagestreamtag resources. If we add support for another kind in the future, we wouldn't want to automatically pick up this behavior for it